internal/local/structure.go

```diff
@@ -230,3 +233,180 @@ func parseServices(rootDir string) ([]ServiceStructure, error) {
 
 	return out, nil
 }
+
+func writeSecrets(rootDir string, secrets *SecretsStructure) error {
+	if secrets == nil {
+		return nil
+	}
+	data, err := MarshalJSON(secrets)
+	if err != nil {
+		return err
+	}
+	return WriteFile(filepath.Join(rootDir, FileSecrets.String()), 0666, bytes.NewReader(data))
+}
+
+func writeEnvironments(rootDir string, environments map[string]map[string]interface{}) error {
+	for env, values := range environments {
+		data, err := MarshalJSON(values)
+		if err != nil {
+			return err
+		}
+		if err = WriteFile(
```

Comment:
[nothing to change here] but with this inline ifs, you can use `:=` and it'll keep it scoped properly (https://play.golang.org/p/WP8C0eMoyL7)